spark/src/main/scala/io/delta/sharing/spark/DeltaSharingClient.scala

```diff
@@ -396,6 +401,14 @@ private[spark] class DeltaSharingRestClient(
       }
     }
 
+  private def getUserAgent(): String = {
+    DeltaSharingRestClient.USER_AGENT + (if (forStreaming) {
+      " StreamingQuery/1.0.0"
```

Comment:
Can the version number be a constant somewhere?

I wonder in what scenario would this version change... not sure if possible to pass this in as a variable somehow

Reply:
I actually don't know what's the best string to put here. So trying to put a feature+version.
This version could change, as we support more parameters(such as Trigger.AvailableNow, which is available in spark 3.3.0, but we are currently at 3.1.1, we are not bumping up the version for other reasons) in streaming.


@chakankardb Thinking about this more, this could be in the same place as what version of dv/cm is supported in the client(once we have them), I hope it to be in a different header, but I don't see a good label for this purpose in the list of headers: https://hc.apache.org/httpcomponents-core-5.2.x/current/httpcore5/apidocs/org/apache/hc/core5/http/HttpHeaders.html.

So maybe we can leverage the user agent header for this purpose, what do you think?

Reply:
Using UA for encoding supported features seems a bit odd. 
Let's discuss it more when we work on it.

Reply:
I feel odd too, sg, let's revisit later.